src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml.cs

```diff
@@ -161,15 +161,26 @@ protected override void OnKeyUp(KeyEventArgs e)
                         if (_versions.SelectedIndex >= 0 && isAVersion)
                         {
                             // Check if the user typed a custom version
-                            if (isUserInputValidNuGetVersionRange)
+                            if (isUserInputValidNuGetVersionRange && userRequestedVersionRange != null && _versions.Items?.CurrentItem != null)
```

Comment:
Looking at line no:159 (copied below), I think we can avoid `userRequestedVersionRange != null` null check because `isUserInputValidNuGetVersionRange` will be `true` only when `userRequestedVersionRange` is non null. Please let me know if I missed something.

```cs
bool isUserInputValidNuGetVersionRange = _versions.SelectedIndex >= 0 ? VersionRange.TryParse(comboboxText, out userRequestedVersionRange) : false;
```